### PR TITLE
ert2 compatibility

### PR DESCRIPTION
**Pull request by ManInFez**

### CODE REVIEW

File: ert_storage/endpoints/records.py [hunk 1]
```diff
@@ -224,6 +225,11 @@ async def post_ensemble_record_matrix(
 
             stream = io.BytesIO(await request.body())
             content = read_array(stream)
+        elif content_type == "application/x-dataframe":
+            stream = io.BytesIO(await request.body())
+            df = pd.read_csv(stream, float_precision="round_trip")
+            content = df.to_numpy()
```
Comment (xjules) [inline ert_storage/endpoints/records.py:231]: Here we can go directly to ArrayFloat by `df.values.tolist()`

File: tests/integration/test_records.py [hunk 2]
```diff
@@ -169,6 +169,42 @@ def test_ensemble_matrix_json(client, simple_ensemble, get, post):
...
+    resp = client.get(
+        f"/ensembles/{ensemble_id}/records/mat",
+        headers={"accept": "application/x-dataframe"},
+    )
+    stream = io.BytesIO(resp.content)
+    df = pd.read_csv(stream, float_precision="round_trip")
+
+    assert_array_equal(df.values, data.values)
+    assert (df.columns.values == data.columns.values).all
+    assert (df.index.values == data.index.values).all
```
Comment (dotfloat) [inline tests/integration/test_records.py:205]:
```suggestion
    assert_array_equal(df.columns.values, data.columns.values)
    assert_array_equal(df.index.values, data.index.values)
```

**Comment by dotfloat** [comment]
Looks good otherwise. Squish and I'll approve.

**Comment by ManInFez** [comment]
The commits are created like this, as there are many individual things which are not strictly related

**Review by dotfloat — APPROVED** [review]

**Pull request merged**